File: Data Structures & Algorithms/meeting-schedule-ii/submission-3.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        TreeMap<Integer, Integer> vals = new TreeMap<>();
        for(Interval i: intervals) {
            vals.put(i.start,vals.getOrDefault(i.start, 0)+ 1);
            vals.put(i.end, vals.getOrDefault(i.end, 0)-1);
        }

        int ans = 0, curr = 0;

        for(int key: vals.keySet()) {
            curr+=vals.get(key);
            ans = Math.max(curr, ans);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-3.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> dp(n, -1);
        return helper(dp, 0, n);
    }

    int helper(vector<int> &dp, int curr, int n) {
        if(curr == n) return 1;
        if(curr > n) return 0;
        if(dp[curr] != -1)
            return dp[curr];
        int ans = helper(dp, curr+1, n);
        ans+= helper(dp, curr+2,n);

        return dp[curr] = ans;
    }
};
